runtime-transaction/src/instruction_processor.rs

```diff
@@ -0,0 +1,62 @@
+use {
+    crate::program_id_flags::ProgramIdFlag,
+    agave_transaction_view::static_account_keys_frame::MAX_STATIC_ACCOUNTS_PER_PACKET as FILTER_SIZE,
+    core::borrow::Borrow, solana_pubkey::Pubkey, solana_sdk::transaction::TransactionError,
+    solana_svm_transaction::instruction::SVMInstruction,
+};
+
+pub trait InstructionProcessor<T>: Default {
+    type OUTPUT;
+
+    /// Process instructions individually given a `flag` that is derived from
+    /// the program id.
+    fn process_instruction(
+        &mut self,
+        flag: &T,
+        instruction_index: usize,
+        instruction: &SVMInstruction,
+    ) -> Result<(), TransactionError>;
+
+    /// Finalize the processor and return the output.
+    fn finalize(self) -> Self::OUTPUT;
+}
+
+pub struct CachedInstructionProcessor<T, P: InstructionProcessor<T>> {
+    flags: [Option<T>; FILTER_SIZE as usize],
```

Comment:
why number of flags are the same as number of static accounts?

Reply:
because their program id flags. check out the caches of the other instruction processing we do - we are identifying which program_ids (which are static keys) correspond to which kind of ix